src/Linter/Sniff/Visitor/ConfigNoCacheVisitor.php

```diff
@@ -0,0 +1,58 @@
+<?php declare(strict_types=1);
+
+namespace Helmich\TypoScriptLint\Linter\Sniff\Visitor;
+
+use Helmich\TypoScriptLint\Linter\Report\Issue;
+use Helmich\TypoScriptLint\Linter\Sniff\ConfigNoCacheSniff;
+use Helmich\TypoScriptParser\Parser\AST\Operator\Assignment;
+use Helmich\TypoScriptParser\Parser\AST\Statement;
+
+class ConfigNoCacheVisitor implements SniffVisitor
+{
+    /** @var Issue[] */
+    private $issues = [];
+
+    /**
+     * @return Issue[]
+     */
+    public function getIssues(): array
+    {
+        return $this->issues;
+    }
+
+    public function enterTree(array $statements): void
+    {
+    }
+
+    public function enterNode(Statement $statement): void
+    {
+         if (!$statement instanceof Assignment) {
+             return;
+         }
+         if ($statement->object->relativeName !== 'no_cache'
+             && !str_ends_with($statement->object->relativeName, '.no_cache')) {
+             return;
+         }
+         if ($statement->value->value !== '0') {
+             $this->issues[] = new Issue(
+                $statement->sourceLine,
+                null,
+                sprintf(
+                    'Setting config.no_cache = 1 is discouraged as it is bad for performance.'
+                        . 'Consider using USER_INT object instead. Found in path: %s',
+                    $statement->object->absoluteName
+                ),
+                Issue::SEVERITY_WARNING,
+                ConfigNoCacheSniff::class
+            );
+         }
```

Comment:
There's a bit of potential for false-positives, here. 🤔  Right now, this sniff would alert on any assignment of a `no_cache` property anywhere, while strictly speaking, it should only alert on an assignment on `config.no_cache` or `{page}.config.no_cache` (with `{page}` being any TS object of type `PAGE`).

I acknowledge that fixing this would lead to a significantly more complex solution (one would basically have to keep track of the current scope while descending the AST) and I'm not sure if this isn't more of a hypothetical concern.

What's your opinion on this, @tmotyl?

Reply:
I thought about it and considered it low risk:
- the no-cache property is used only in one context in the core TS
- we’re yelding a warning and not error

I also didint know how to detect object type if the object is not declared in the same file.

In my pov a pragmatic approach would be to get it as is and if somebody complain about false positives - then tune it, having real case at hand.

Reply:
> I also didint know how to detect object type if the object is not declared in the same file.

True, that would have been an additional challenge... 😓 

> In my pov a pragmatic approach would be to get it as is and if somebody complain about false positives - then tune it, having real case at hand.

I can live with that. :+1: